solidity/interfaces/ITransformer.sol

```diff
@@ -50,6 +50,34 @@ interface ITransformer {
     view
     returns (uint256 amountDependent);
 
+  /**
+   * @notice Calculates how many dependent tokens are needed to transform to the expected
+   *         amount of underlying
+   * @dev This function must be unaware of context. The returned values must be the same,
+   *      regardless of who the caller is
+   * @param dependent The address of the dependent token
+   * @param expectedUnderlying The expected amounts of underlying tokens
+   * @return neededDependent The amount of dependent needed
+   */
+  function calculateNeededToTransformToUnderlying(address dependent, UnderlyingAmount[] calldata expectedUnderlying)
+    external
+    view
+    returns (uint256 neededDependent);
```

Comment:
There might be cases where this will be a "best effort" I think. For example: If we here ask for a LP 4626 and send underlying amounts that do not match the "weight" of the LP, what should happen there?

I feel LPs are a really special case on everything thoo.

Reply:
Agreed, this will always be a "best effort" action